packages/devtools_flutter/lib/src/config.dart

```diff
@@ -10,28 +11,119 @@ import 'screen.dart';
 
 /// Top-level configuration for the app.
 @immutable
-class Config {
-  /// The routes that the app exposes.
-  Map<String, WidgetBuilder> get routes {
-    final routeToBuilder = <String, WidgetBuilder>{};
-    for (var key in _routeToTabs.keys) {
-      routeToBuilder[key] =
-          (BuildContext context) => DevToolsScaffold(tabs: _routeToTabs[key]);
+class DevToolsApp extends StatefulWidget {
+  @override
+  State<DevToolsApp> createState() => DevToolsAppState();
+}
+
+/// Initializer for the [FrameworkCore] and the app's navigation.
+///
+/// This manages the route generation, and marshalls URL query parameters into
+/// flutter route parameters.
+// TODO(https://github.com/flutter/devtools/issues/1146): Introduce tests that
+// navigate the full app.
+class DevToolsAppState extends State<DevToolsApp> {
+  @override
+  void initState() {
+    super.initState();
+    service.FrameworkCore.init(WidgetsBinding.instance.window.defaultRouteName);
+  }
+
+  /// Generates routes, separating the path from URL query parameters.
+  Route _generateRoute(RouteSettings settings) {
+    final uri = Uri.parse(settings.name);
+    final path = uri.path;
+
+    if (_routes.containsKey(path)) {
+      return MaterialPageRoute(settings: settings, builder: _routes[path]);
     }
-    return routeToBuilder;
+    // Return a page not found.
+    return MaterialPageRoute(
+      settings: settings,
+      builder: (BuildContext context) {
```

Comment:
Optional: can we use the single expression statement here instead of block? i.e :`(BuildContext context) => DevToolsScaffold.withChild(...)`

Or is there any convention / guide on using single / block statement?

Reply:
Code readability is how we draw a line here.

The one-liner syntax can be easier to read when the contents are simple, but when the code inside is more complicated, even if it's one line, the code is generally easier to read when wrapped in curly braces.

https://dart.dev/guides/language/effective-dart/usage#consider-using--for-simple-members

Reply:
Ah I see, thanks Dave!